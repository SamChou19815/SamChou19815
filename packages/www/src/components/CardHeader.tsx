import React from 'react';

type Props = { readonly title: string; readonly subheader?: string };

export default function CardHeader({ title, subheader }: Props): JSX.Element {
  return (
    <div className="p-4 pb-0 last:pb-4">
      <h4>{title}</h4>
      {subheader && <small>{subheader}</small>}
    </div>
  );
}
