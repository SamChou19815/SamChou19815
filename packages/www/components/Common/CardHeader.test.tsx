import React from 'react';

import renderer from 'react-test-renderer';

import CardHeader from './CardHeader';

it('Simple CardHeader matches snapshot.', () => {
  const tree = renderer.create(<CardHeader title="t" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Simple CardHeader with subheader matches snapshot.', () => {
  const tree = renderer.create(<CardHeader title="t" subheader="s" />).toJSON();
  expect(tree).toMatchSnapshot();
});
