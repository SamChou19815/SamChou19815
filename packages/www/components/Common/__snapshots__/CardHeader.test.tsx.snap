// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple CardHeader matches snapshot. 1`] = `
<div
  className="card__header"
>
  <div
    className="avatar"
  >
    <div
      className="avatar__intro"
    >
      <h4
        className="avatar__name"
      >
        t
      </h4>
    </div>
  </div>
</div>
`;

exports[`Simple CardHeader with subheader matches snapshot. 1`] = `
<div
  className="card__header"
>
  <div
    className="avatar"
  >
    <div
      className="avatar__intro"
    >
      <h4
        className="avatar__name"
      >
        t
      </h4>
      <small
        className="avatar__subtitle"
      >
        s
      </small>
    </div>
  </div>
</div>
`;
